iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -250,4 +305,74 @@ static void overlayTableProperties(Configuration configuration, TableDesc tableD
     // this is an exception to the interface documentation, but it's a safe operation to add this property
     props.put(InputFormatConfig.TABLE_SCHEMA, schemaJson);
   }
+
+  /**
+   * Recursively collects the column names from the predicate.
+   * @param node The node we are traversing
+   * @param columns The already collected column names
+   */
+  private void columns(ExprNodeDesc node, Collection<String> columns) {
```

Comment:
why is this a `Collection` - do we need to collect the same column multiple times?

Reply:
Based on my tests the `addDynamicSplitPruningEdge` will be called for every column and the `ExprNodeDesc` only refers to a single column, so in theory we can handle multiple columns and everything, but we do not have to optimize for multi-column case. Using `Set` would work as well, but it is more strict and I wanted to keep things as general as possible.

Reply:
In the end moved to `Set` since it was needed to make sure than only a single column is there in the pruning expression